config/default/forms/app/replace_user.properties.json

```diff
@@ -0,0 +1,9 @@
+{
+    "icon": "icon-death-general",
+    "title": "Replace User",
+    "context": {
+        "person": true,
+        "place": false,
+        "expression": "user._id === contact._id && ['chw', 'chw_supervisor'].includes(user.role) && (!contact.user_for_contact || !contact.user_for_contact.replace)"
```

Comment:
@dianabarsan @m5r the logic in here for hiding the form if the user is already replaced (e.g. if the user is offline and replaced, but have not synced, they should not see the form) is broken for cases where there are multiple users associated with the same contact (since once one user is replaced, and the contact doc changes are synced, none of the other users will be able to see the form).

It would not be hard to make this logic user-specific by adding `|| !contact.user_for_contact.replace[username]` but currently we do not have access to the `username` in these expressions (or to any other data that would uniquely identify the user...).  One option would be to update the `xml-forms` service and add the `username` to the [set of data available](https://github.com/medic/cht-core/blob/5d2a685c601161274fad14451dd7d9dcccd0f798/webapp/src/ts/services/xml-forms.service.ts#L86).  It does not seem like that would be too hard, but I am not sure it is needed for this MVP implementation....

Reply:
I'm curious if we indeed want this form to be available to `chw` users. This means that any user, at any time, can submit this form and replace themselves?

Reply:
Are there any precautions in the form to make sure that the current device has the correct SIM that would receive the token login link?

Reply:
The user replace happens completely offline so `chw` users must have access to the form.  
To prevent users from submitting the form and replacing themselves, the form is configured with a code that only supervisors *should* know. Without that code, the form cannot do anything.

Reply:
No, the form doesn't validate the phone's SIM card, it relies on the user's phone number being correct. I need to double check first but I believe the phone number field is pre-filled with the user's phone number but editable in case the replacing CHW has a different phone number

Reply:
We built this feature with certain requirements and this is one of them. The form can't be used without the code and the code can't be known if it wasn't configured in the first place. Plus, the form can always be removed from the deployment config if it's not relevant for a partner

Reply:
Just wanted to add a few additional details/clarifications:  

- The contact for the new user _must_ have a _valid_ phone number because the new user is created with token login enabled.  The `create_user_for_contacts` functionality will produce an error if a phone number is not set on the contact.
- The default `replace_user` form we have created pulls the phone number for the new contact from the existing user's `user-settings`. If no phone number exists there, it requires the user to input one before submitting the form. These are _implementation details_ of our `replace_user` form and are not required for the `create_user_for_contacts` functionality. Projects can write their own custom form and pull the phone number from wherever they want.
- Being able to swap SIM cards in the phone when the user is replaced offline was one of the design elements that we wanted (so the new user does not have to be locked down to the phone number of the old user).